packages/core/src/odata/common/filter/filterable.ts

```diff
@@ -67,3 +72,40 @@ export function or<EntityT extends EntityBase>(
 // Export function not<EntityT extends Entity>(expression: Filterable<EntityT>): Filterable<EntityT> {
 //   Return new FilterList([], expressions);
 // }
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @experimental This is experimental and is subject to change. Use with caution.
+ */
+export function any<FieldT extends FieldType, LinkedEntityT extends EntityBase>(
+  filter: Filter<LinkedEntityT, FieldT>
+): FilterWithLambdaOperator<LinkedEntityT, FieldType> {
+  return { filter, lambdaOperator: 'any' };
+}
```

Comment:
I think we should use `FilterLambdaExpression` here and we should get rid of the additional `FilterWithLambdaOperator` type.

```suggestion
  ...filterables: Filterable<EntityT>[]
): FilterLambdaExpression<EntityT, FieldType > {
  return new FilterLambdaExpression(filterables, 'any'); // see my comments on the constructor
}
```